Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -117,6 +117,92 @@ class FileLogSource : LogSource {
     }
 }
 
+class EventLogLogSource : LogSource {
+    [String] $EventLogName
+    [String] $EventLogSource
+    [int64] $StartEventIdx
+
+    EventLogLogSource($Session, $EventLogName, $EventLogSource) {
+        $This.Session = $Session
+        $this.EventLogName = $EventLogName
+        $this.EventLogSource = $EventLogSource
+        $this.StartEventIdx = $this.GetLatestEventIdx() + 1
+    }
+
+    [CollectedLog[]] GetContent() {
+        $LogGetterBody = {
+            Param([Parameter(Mandatory = $true)] [string] $LogName,
+                  [Parameter(Mandatory = $true)] [string] $LogSource,
+                  [Parameter(Mandatory = $true)] [int64] $StartEventIdx,
+                  [Parameter(Mandatory = $true)] [int64] $EndEventIdx)
+
+            $Name = "$LogSource - $LogName event log"
+
+            try {
+                $Content = Get-EventLog `
+                    -LogName $LogName `
+                    -Source $LogSource `
+                    -Index ($StartEventIdx..$EndEventIdx) | Format-Table -Wrap | Out-String
+            } catch {
+                return @{
+                    Name = $Name
+                    Err = "event log retrieval error: $($_.Exception.Message)"
+                }
+            }
+
+            # Check the index range after getting content from event log, so that we get a chance
+            # to catch any other exceptions thrown by Get-EventLog.
+            if ($StartEventIdx -gt $EndEventIdx) {
```

Comment:
i think we should check -ge instead of -gt. This way we won't omit one-element changes